daemon/loop.go

```diff
@@ -21,40 +21,74 @@ type LoopVars struct {
 	GitPollInterval      time.Duration
 	RegistryPollInterval time.Duration
 	syncSoon             chan struct{}
-	initSyncSoon         sync.Once
+	pollImagesSoon       chan struct{}
+	initOnce             sync.Once
+}
+
+func (loop *LoopVars) ensureInit() {
+	loop.initOnce.Do(func() {
+		loop.syncSoon = make(chan struct{}, 1)
+		loop.pollImagesSoon = make(chan struct{}, 1)
+	})
 }
 
 // Loop for potentially long-running stuff. This includes running
 // jobs, and looking for new commits.
 
 func (d *Daemon) Loop(stop chan struct{}, wg *sync.WaitGroup, logger log.Logger) {
 	defer wg.Done()
-	pollGit := time.NewTimer(d.GitPollInterval)
+	gitPollTimer := time.NewTimer(d.GitPollInterval)
+	// This can be called any time to push the next git poll out by
+	// d.GitPollInterval. It must _at least_ be called when the timer
+	// fires, so that the polling continues.
 	resetGitPoll := func() {
```

Comment:
There are some functions defined as methods on the daemon (e.g. `doSync`) and others that are inlined

Why have you inlined this function and the two below? Personally, I think it bloats the Loop method.

I would prefer separate methods / class for the git timer stuff, and move `pullThen` as a proper Daemon method (because of it's use of `d.Checkout...`).

But obviously, this is just a style issue nit. No biggie.